presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentProvider.java

```diff
@@ -32,7 +33,12 @@
 
     protected EnvironmentProvider(List<EnvironmentExtender> bases)
     {
-        this.bases = requireNonNull(bases, "bases is null");
+        this.bases = copyOf(requireNonNull(bases, "bases is null"));
+    }
+
+    protected EnvironmentProvider(EnvironmentExtender... bases)
+    {
+        this.bases = copyOf(requireNonNull(bases, "bases is null"));
```

Comment:
call `this(...)` that way there is one true constructor and several convenience constructors. Then move this constructor up, so they are ordered by usage (used lower).